src/Service.Tests/OpenApiDocumentor/ParameterValidationTests.cs

```diff
@@ -0,0 +1,198 @@
+                        Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name is "$first" && param.Schema.Type is "integer"));
+                        Assert.IsTrue(operation.Parameters.Any(param => param.In is ParameterLocation.Query && param.Name is "$after" && param.Schema.Type is "string"));
+                    }
+                    else
+                    {
+                        // No query parameters for other OperationTypes. 
+                        Assert.IsFalse(operation.Parameters.Any(param => param.In is ParameterLocation.Query));
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Validates that Query parameters are correctly generated for Stored Procedures.
+        /// Query Parameters for stored procedure will be generated for irrespective of the Operation Type.
+        /// If a parameter default value is not provided in the config, it will marked as required.
+        /// </summary>
+        /// <param name="entityName"></param>
+        /// <param name="objectName"></param>
+        /// <returns></returns>
+        [DataTestMethod]
+        [DataRow("CountBooks", "count_books", DisplayName = "StoredProcedure with no parameters")]
+        [DataRow("UpdateBookTitle", "update_book_title", DisplayName = "StoredProcedure with parameters")]
+        public async Task ValidateQueryParametersForStoredProcedures(string entityName, string objectName)
+        {
+            Dictionary<string, object> parameterDefaults = null;
+            if (entityName is "UpdateBookTitle")
+            {
+                // Adding a parameter default value.
+                parameterDefaults = new Dictionary<string, object> { { "title", "Test Title" } };
+            }
+
+            Entity entity = new(
+                Source: new(Object: objectName, EntitySourceType.StoredProcedure, parameterDefaults, null),
+                GraphQL: new(Singular: null, Plural: null, Enabled: false),
+                Rest: new(Methods: EntityRestOptions.DEFAULT_SUPPORTED_VERBS),
+                Permissions: OpenApiTestBootstrap.CreateBasicPermissions(),
+                Mappings: null,
+                Relationships: null);
+
+            Dictionary<string, Entity> entities = new()
+            {
+                { entityName, entity }
+            };
+
+            RuntimeEntities runtimeEntities = new(entities);
+            OpenApiDocument openApiDocument = await OpenApiTestBootstrap.GenerateOpenApiDocument(
+                runtimeEntities: runtimeEntities,
+                configFileName: CUSTOM_CONFIG,
+                databaseEnvironment: MSSQL_ENVIRONMENT);
+
+            OpenApiPathItem pathItem = openApiDocument.Paths.First().Value;
+            foreach (OpenApiOperation operation in pathItem.Operations.Values)
+            {
+                if (entityName is "UpdateBookTitle")
```

Comment:
For string data types, should use the string specific comparison function - `.Equals()`.